niworkflows/utils/images.py

```diff
@@ -3,6 +3,16 @@
 import numpy as np
 
 
+def as_canonical(img):
+    """Drop rotation w.r.t. cardinal axes of input image."""
+    img = nb.as_closest_canonical(img)
+    zooms = list(img.header.get_zooms()[:3])
+    newaff = np.diag(zooms + [1])
+    rot = newaff[:3, :3].dot(np.linalg.inv(img.affine[:3, :3]))
+    newaff[:3, 3] = rot.dot(img.affine[:3, 3])
```

Comment:
This can be simplified. Rendering in a more readable form:

```Python
rot = newaff[:3, :3] @ (img.affine[:3, :3] ** -1)
newaff[:3, 3] = rot @ img.affine[:3, 3]
```

Substituting:

```Python
newaff[:3, 3] = newaff[:3, :3] @ (img.affine[:3, :3] ** -1) @ img.affine[:3, 3]
newaff[:3, 3] = newaff[:3, :3]
```

```suggestion
```

Reply:
Tried the `img.affine[:3, :3] ** -1` but SVD does not converge